### PR TITLE
add pagination

### PR DESCRIPTION
**Pull request by rgvillanueva28**
PR for #41

### CODE REVIEW

File: src/components/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+interface PaginationInterface {
+  page: number;
+  length: number;
+  size: number;
+  setPage: Dispatch<SetStateAction<number>>;
+}
+
+function Pagination({ page, setPage, length, size }: PaginationInterface) {
+  console.log(length / size);
```
Comment (jnebab) [inline src/components/Pagination.tsx:12]: please remove this console log

File: src/components/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          disabled={page <= 1}
+          onClick={() => setPage(page - 1)}
+        >
+          <BsArrowLeft size={24} />
+          <span>Prev</span>
+        </button>
+        <button
+          className=" flex flex-row items-center space-x-3 rounded-r-md border-l border-white bg-gray-700 py-2 px-4 text-white hover:bg-gray-600 disabled:bg-gray-500 disabled:cursor-not-allowed"
+          onClick={() => setPage(page + 1)}
+          disabled={length / size <= page}
```
Comment (jnebab) [inline src/components/Pagination.tsx:32]: you can use the `directories.hasNextPage` bool to replace this condition.

File: src/components/Pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  setPage: Dispatch<SetStateAction<number>>;
+}
+
+function Pagination({ page, setPage, length, size }: PaginationInterface) {
+  console.log(length / size);
+  return (
+    <div className="mt-5 flex flex-col items-center justify-center">
+      <p className="mb-2 text-center">
+        Showing <strong>{page * size - (size - 1)}</strong> to{" "}
+        <strong>{length >= size * page ? size * page : length}</strong> of{" "}
```
Comment (jnebab) [inline src/components/Pagination.tsx:17]: You can also use `hasNextPage` here.

**Comment by jnebab** [comment]
Hi @rgvillanueva28, we thank you for your contribution. Kindly check the feedback. Thank you.

**Comment by jnebab** [comment]
lgtm

File: src/components/Pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        <button
+          className="flex flex-row items-center space-x-3 rounded-l-md bg-gray-700 py-2 px-4 text-white hover:bg-gray-600 disabled:cursor-not-allowed disabled:bg-gray-500"
+          disabled={page <= 1}
+          onClick={() => setPage(page - 1)}
+        >
+          <BsArrowLeft size={24} />
+          <span>Prev</span>
+        </button>
+        <button
+          className=" flex flex-row items-center space-x-3 rounded-r-md border-l border-white bg-gray-700 py-2 px-4 text-white hover:bg-gray-600 disabled:cursor-not-allowed disabled:bg-gray-500"
```
Comment (jnebab) [inline src/components/Pagination.tsx:36]:
```suggestion
          className=" flex flex-row items-center space-x-3 rounded-r-md border-l border-white bg-gradient-to-r from-purple-900 to-violet-500 py-2 px-4 text-white hover:bg-gray-600 disabled:cursor-not-allowed disabled:bg-gray-500"
```

**Review by jnebab — CHANGES_REQUESTED** [review]

**Review by warengonzaga — APPROVED** [review]
LGTM ✌️

**Comment by warengonzaga** [comment]
Found an issue @rgvillanueva28 not sure why it shows like this?
image

cc: @jnebab @kquirapas

**Comment by warengonzaga** [comment]
> Any error messages? @warengonzaga

I don't see any error messages but with every turn of pages, it has 2 missing items.

**Comment by jnebab** [comment]
> Found an issue @rgvillanueva28 not sure why it shows like this? image
> 
> cc: @jnebab @kquirapas

This actually make sense. Since the pagination is only getting 10 items at a time and the grid is in 3 columns. so for 10 items, it will have 3 columns and 4 rows with the 4th row with one item only. Maybe we can make it 12 items per page to fill  it up correctly

**Comment by warengonzaga** [comment]
> > Found an issue @rgvillanueva28 not sure why it shows like this? image
> > cc: @jnebab @kquirapas
> 
> This actually make sense. Since the pagination is only getting 10 items at a time and the grid is in 3 columns. so for 10 items, it will have 3 columns and 4 rows with the 4th row with one item only. Maybe we can make it 12 items per page to fill it up correctly

Yeah let's make it 12 to fill up the missing 2 items.

**Comment by jnebab** [comment]
@rgvillanueva28, please update it to 12 per page. Thanks for the help :)

**Review by jnebab — APPROVED** [review]
lgtm.

**Review by warengonzaga — APPROVED** [review]
LGTM

**Pull request merged**